src/Dataflow/Rendering/Shaders/EnvLightNode/shader.frag.glsl

```diff
@@ -0,0 +1,125 @@
+
+// sample implementation
+void main() {
+    vec3 tc = getPerVertexTexCoord();
+    // discard non opaque fragment
+    vec4 bc = getBaseColor( material, tc );
+    if ( toDiscard( material, bc ) ) discard;
+
+#ifdef GLTF_MATERIAL_INTERFACE
+    // Experiment on a new GLSL/Material interface allowing more efficient PBR and composition
+
+    // Compute the normal closure. Do normal map and capture state about normal and tangent space
+    NormalInfo nrm_info = getNormalInfo( material.baseMaterial, tc );
+    vec3 view = normalize( in_viewVector );
+
+    MaterialInfo bsdf_params;
+    BsdfInfo layers;
+
+    if ( getSeparateBSDFComponent( material,
+                                   tc,
+                                   view,
+                                   nrm_info,
+                                   bsdf_params,
+                                   layers ) == 1 ) {
+
+        vec2 size = textureSize( amb_occ_sampler, 0 ).xy;
+        vec3 ao   = texture( amb_occ_sampler, gl_FragCoord.xy / size ).rgb;
+
+        float r  = GGXroughness( bsdf_params );
+        // diffuse irradiance map
+        vec4 nrm = vec4( nrm_info.n, 1 );
+        vec3 irradiance = vec3(
+            dot( nrm, redShCoeffs * nrm ),
+            dot( nrm, greenShCoeffs * nrm ),
+            dot( nrm, blueShCoeffs * nrm )
+        );
+        layers.f_diffuse *= irradiance;
+
+        // Specular envmap
+        vec3 rfl = reflect( -view, nrm_info.n );
+        layers.f_specular *=  textureLod( envTexture, rfl, r * numLod ).rgb ;
+
+#ifdef CLEARCOAT_LAYER
+        // clearcoat envmap
+        r = pow(bsdf_params.clearcoat.intrough.y, 0.5);
+        layers.f_clearcoat *= textureLod( envTexture, rfl, r * numLod ).rgb ;
+#endif
+
+#ifdef SHEEN_LAYER
+        r = pow(bsdf_params.sheen.sheenColorRough.a, 0.5);
+        layers.f_sheen *=  textureLod( envTexture, rfl, r * numLod ).rgb ;
+#endif
+
+        bc.rgb = combineLayers( bsdf_params, layers, nrm_info, view );
+
+        bc.rgb *= ao * envStrength;
+    } else {
+        bc.rgb = vec3(0);
+    }
+
+#else
+    vec4 normalWorld = vec4(
+        getNormal(
+            material, tc, getWorldSpaceNormal(), getWorldSpaceTangent(), getWorldSpaceBiTangent() ),
+        1 );
+
+    vec3 diffuse;
+    vec3 specular;
+    vec3 view = normalize( in_viewVector );
+    vec3 rfl  = reflect( -view, normalWorld.xyz );
+
+    if ( getSeparateBSDFComponent(
+             material, getPerVertexTexCoord(), rfl, view, normalWorld.xyz, diffuse, specular ) ==
+         1 )
+    {
+        vec2 size = textureSize( amb_occ_sampler, 0 ).xy;
+        vec3 ao   = texture( amb_occ_sampler, gl_FragCoord.xy / size ).rgb;
+        bc.rgb    = diffuse;
+        vec3 irradiance = vec3(
+            dot( normalWorld, redShCoeffs * normalWorld ),
+            dot( normalWorld, greenShCoeffs * normalWorld ),
+            dot( normalWorld, blueShCoeffs * normalWorld )
+        );
+        bc.rgb *= irradiance / OneOverPi;
```

Comment:
`irradiance / OneOverPi` or, similar, `irradiance * Pi` (The constant `OneOverPi` is defined here, not the constant Pi ^^). But this is only because our implementation of the Blinn-Phong bsdf (or non PBR bsdf) gives a diffuse reflectance divided by `Pi`. The GLTF path, which is PBR, does not need this.

Reply:
After a second thought, there were some double multiplication by `OneOverPi` that degraded the non PBR path. Shader is fixed, hereis envmap only shading of our `glossy_spheres.dae` test scene.
<img width="793" alt="glossySpheres" src="https://user-images.githubusercontent.com/11328578/229874422-8ae0c6de-8393-4ed2-82ed-f7b90fcc37e9.png">